runtime/cpuset/cpuset_builder_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cpuset
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCPUSet(t *testing.T) {
+	cases := []struct {
+		name           string
+		cpus           []int
+		mems           []int
+		cpuRanges      []_range
+		memRanges      []_range
+		expectedCPUSet CPUSet
+	}{
+		{
+			name: "single cpu only",
```

Comment:
Worth adding cases for empty cpus/empty cpuRanges?

Reply:
Done